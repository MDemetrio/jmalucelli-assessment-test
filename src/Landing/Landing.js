import React, { Component } from 'react'
import styled from "styled-components";
import lineChart from '../assets/line-chart.svg';
import { Button } from '../components';
import SineWave from '../components/SineWave';

const Container = styled.main`
  background-image: linear-gradient(#8386f0, #a36cf4);
  min-height: 100vh;
  min-width: 100vw;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;

const Content = styled.section`
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 16vh;
  color: white;
  text-align: center;
  font-size: calc(8px + 2vmin);
  @media (min-width: 600px) {
    margin-top: 0;
  }
`;

const LineChartIcon = styled.img`
  height: 6vmin;
  margin-bottom: 10%;
`;

const H1 = styled.h1`
  font-weight: bold;
  letter-spacing: 1px;
`;

const H2 = styled.h2`
  font-family: "Open Sans";
  font-size: 1rem;
  opacity: 0.7;
  margin: 0px;
`;

export default class Landing extends Component {
  render() {
    return (
      <Container>
        <SineWave />
        <Content>
          <LineChartIcon src={lineChart} alt="line-chart" />
          <H1>Cotação de seguros</H1>
          <H2>Solução inovadora da líder de mercado</H2>
          <Button>Iniciar</Button>
        </Content>
      </Container>
    )
  }
}
