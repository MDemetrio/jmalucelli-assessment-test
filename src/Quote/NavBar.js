import React from 'react'
import styled from "styled-components";
import { animated } from 'react-spring'
import lineChart from '../assets/line-chart.svg';

const Header = styled(animated.header)`
  background-image: linear-gradient(to left, #8386f0, #a36cf4);
  height: 14vh;
  padding: 0px 10vw;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;

  @media (min-width: 600px) {
    padding: 0px 30vw;
    height: 10vh;
  }
`;

const InfoContainer = styled(animated.div)`
  display: flex;
  align-items: center;
`;

const LineChartIcon = styled.img`
  height: 5vmin;
`;

const TitleContainer = styled(animated.div)`
  margin-left: 20px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  color: #fff;
  font-family: 'Montserrat';
  font-weight: normal;
`

const Title = styled.p`
  font-size: calc(6px + 2vmin);
  margin: 0px;
`
const Avatar = styled(animated.img)`
  height: calc(24px + 4vmin);
  width: calc(24px + 4vmin);
  vertical-align: middle;
  border-radius: 50%;
`;

const NavBar = ({ style }) =>
  (
    <Header style={style}>
      <InfoContainer>
        <LineChartIcon src={lineChart} alt="line-chart-icon" aria-hidden={true} />
        <TitleContainer style={{ opacity: style.opacity }}>
          <Title>Nova cotação</Title>
          <Title>#0980</Title>
        </TitleContainer>
      </InfoContainer>
      <Avatar style={{ opacity: style.opacity }} src="https://source.unsplash.com/collection/1972462" alt="user-avatar" />
    </Header>

  )
export default NavBar;