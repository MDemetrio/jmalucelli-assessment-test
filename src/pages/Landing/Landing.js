import React from 'react'
import styled from "styled-components";
import PropTypes from 'prop-types';
import lineChart from '../../assets/line-chart.svg';
import { Button } from '../../components';
import SineWave from '../../components/SineWave';
import arrowFoward from '../../assets/arrow-foward.svg';

const Container = styled.main`
  background-image: linear-gradient(#8386f0, #a36cf4);
  min-height: 100vh;
  min-width: 100vw;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;

const Content = styled.section`
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 16vh;
  color: white;
  text-align: center;
  font-size: calc(8px + 2vmin);
  @media (min-width: 600px) {
    margin-top: 0;
  }
`;

const LineChartIcon = styled.img`
  height: 6vmin;
  margin-bottom: 10%;
`;

const H1 = styled.h1`
  font-weight: bold;
  letter-spacing: 1px;
`;

const H2 = styled.h2`
  font-family: "Open Sans";
  font-size: 1rem;
  opacity: 0.7;
  margin: 0px;
`;


const Landing = (props) => (
  <Container style={props.style}>
    <SineWave />
    <Content>
      <LineChartIcon src={lineChart} alt="line-chart-icon" aria-hidden={true} />
      <H1>Cotação de seguros</H1>
      <H2>Solução inovadora da líder de mercado</H2>
      <Button icon={arrowFoward} onClick={(e) => {
        e.preventDefault();
        props.history.push({ pathname: '/quote' })
      }}>
        Iniciar
        </Button>
    </Content>
  </Container>
);

Landing.propTypes = {
  history: PropTypes.object.isRequired,
}

export default Landing;
