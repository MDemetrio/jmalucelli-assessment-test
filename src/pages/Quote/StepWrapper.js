import React from 'react';
import styled from "styled-components";


const StepContainer = styled.div`
  display: flex;
  align-items: center;
  padding: 16px 0;
  color: black;
  font-size: calc(8px + 2vmin);
  font-weight: bold;
`;

const Step = styled.div`
  height: 0;
  width: 22px;
  font-size: 1rem;
  padding-bottom: 20px;
  border: 2px solid #8e84ee;
  text-align: center;
  color: #8e84ee;
  margin-right: 20px;
`;

const StepWrapper = (props) => (
    <StepContainer>
        <Step>1</Step>
        Buscar por CNPJ ou empresa
    </StepContainer>
)


export default StepWrapper;