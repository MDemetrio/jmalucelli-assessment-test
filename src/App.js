import React, { Component } from 'react';
import { Switch, Route, Redirect, BrowserRouter as Router } from 'react-router-dom';
import Quote from './pages/Quote';
import Landing from './pages/Landing';

class App extends Component {
  
  componentDidMount() {
    localStorage.setItem('ACCESS-TOKEN', 23456789)
  }

  render() {
    return (
      <Router>
        <Switch>
          <Route exact path="/" component={Landing} />
          <Route exact path="/quote" component={Quote} />
          <Redirect to="/" />
        </Switch>
      </Router>
    );
  }
}

export default App;
