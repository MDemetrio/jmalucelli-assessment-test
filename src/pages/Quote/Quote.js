import React, { Component } from 'react'
import styled from "styled-components";
import { Spring } from 'react-spring';
import NavBar from './NavBar';
import StepWrapper from './StepWrapper';
import { Button } from '../components';
import arrowFowardWhite from '../assets/arrow-foward-white.svg';
import checkCircle from '../assets/check-circle.svg';
import spinner from '../assets/spinner.svg';
import MaskedInput from 'react-text-mask'
import { getQuoteByCnpj } from '../services/quoteService';

const Container = styled.main`
  height: 100vh;
  width: 100vw;
  display: flex;
  flex-direction: column;
  background-color: #f2f6f8;
`;

const Content = styled.section`
  width: 100vw;
  display: flex;
  flex-direction: column;
  align-items: center;
  color: white;
  text-align: center;
  font-size: calc(8px + 2vmin);
`;

const InputLabel = styled.label`
  margin-left: 16px;
  margin-top: 8px;
  align-self: flex-start;
  color: #9ca5ac;

`;

const Input = styled.input`
    padding: 16px 0px;
    border: 0;
    font-size: 1.4rem;
    color: #505969;
    :focus{
    outline:0;
  }
`;

const FieldContainer = styled.div`
  width: 100%;
  background-color: white;
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 6vh 0px;

  @media (min-width: 600px) {
    width: 400px;
  }
`;

const InputContainer = styled.div`
    width: 90%;
  display: flex;
  flex-direction: row;
  align-items: center;
`;

const FieldValidityIcon = styled.img`
  height: 18px;
`;

export default class Quote extends Component {

  state = {
    cnpj: '',
    formValid: false,
    isFetching: false
  }

  handleFieldChange = (event) => {
    const value = event.target.value
    const name = event.target.name

    this.setState({ [name]: value, formValid: this.formEl.checkValidity() })
  }

  handleSubmit = (event) => {
    event.preventDefault();
    const cnpj = this.state.cnpj.replace(/\D/g, '');

    if (this.formEl.checkValidity() && !this.state.isFetching) {
      this.setState({ isFetching: true }, () => {
        getQuoteByCnpj(cnpj)
          .then(r => r.text())
          .then(t => {
            alert(t);
            this.setState({ isFetching: false })
          })
          .catch(e => console.log(e));
      });
    }
  }

  render() {
    const { cnpj, formValid, isFetching } = this.state;

    return (
      <Spring
        native
        config={{ tension: 120, friction: 60 }}
        from={{
          minHeight: 100,
          opacity: 0
        }}
        to={{
          minHeight: 10,
          opacity: 1
        }}>
        {style =>
          <Container>
            <NavBar style={{ opacity: style.opacity, minHeight: style.minHeight.interpolate(x => `${x}vh`) }} />
            <form ref={form => this.formEl = form} onSubmit={this.handleSubmit}>
              <Content>
                <StepWrapper />
                <FieldContainer>
                  <InputLabel htmlFor="cnpj">CNPJ / Empresa</InputLabel>
                  <InputContainer>
                    <MaskedInput
                      mask={[/\d/, /\d/, '.', /\d/, /\d/, /\d/, '.', /\d/, /\d/, /\d/, '/', /\d/, /\d/, /\d/, /\d/, '-', /\d/, /\d/]}
                      onChange={this.handleFieldChange}
                      value={cnpj}
                      id="cnpj"
                      name="cnpj"
                      required
                      pattern="\d{2}\.\d{3}\.\d{3}/\d{4}-\d{2}"
                      render={(ref, props) => (
                        <Input ref={ref} {...props} />
                      )}
                    />

                    {formValid &&
                      <FieldValidityIcon src={checkCircle} alt="field-is-valid" />
                    }

                  </InputContainer>
                </FieldContainer>
                <Button type='submit' disabled={isFetching} icon={isFetching ? spinner : arrowFowardWhite} style={{ backgroundColor: isFetching ? '#ddd' : '#32cccc', color: 'white' }}>
                  OK
              </Button>
              </Content>
            </form>
          </Container>
        }
      </Spring>
    )
  }
}
