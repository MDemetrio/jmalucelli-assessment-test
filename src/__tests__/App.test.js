import React from 'react';
import App from '../App';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router';
import Landing from '../Landing';
import Quote from '../Quote';

it('renders without crashing', () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});

describe('App routing', () => {
  it('should render Landing component on "/" route', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/']}>
        <App />
      </MemoryRouter>
    )
    expect(wrapper.containsMatchingElement(Landing)).toBe(true);
  })

  it('should render Quote component on "/quote" route', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/quote']}>
        <App />
      </MemoryRouter>
    )
    expect(wrapper.containsMatchingElement(Quote)).toBe(true);
  })

  it('should render Landing component on undeclared route', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/undeclaredroute']}>
        <App />
      </MemoryRouter>
    )
    expect(wrapper.containsMatchingElement(Landing)).toBe(true);
  })

});