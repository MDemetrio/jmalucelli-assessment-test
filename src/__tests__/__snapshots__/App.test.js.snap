// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Component>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/quote"
    />
    <Redirect
      push={false}
      to="/"
    />
  </Switch>
</Component>
`;
